state/application.go

```diff
@@ -2224,6 +2243,35 @@ func (a *Application) ChangeScale(scaleChange int) (int, error) {
 			Update: bson.D{{"$set", bson.D{{"scale", newScale}}}},
 		}}
 
+		if scaleChange > 0 && len(storageTags) > 0 {
+			// Since len(storageTags) will always equal to one at this step,
+			// the unit OrderedId will be equal to a.doc.UnitCount.
+			unitName := a.doc.Name + "/" + strconv.Itoa(a.doc.UnitCount)
+
+			// The operations return should equal to a.insertCAASUnitOps but
+			// use different OrderedId check since the desired scale
+			// number haven't been updated.
+			insertUnitOps, err := a.insertCAASUnitOps(
+				UpsertCAASUnitParams{
+					AddUnitParams: AddUnitParams{
+						UnitName:      &unitName,
+						AttachStorage: storageTags,
+					},
+				},
+				func() error {
+					if ps := a.ProvisioningState(); ps != nil && ps.Scaling && a.doc.UnitCount != ps.ScaleTarget+1 {
+						return errors.NotAssignedf("Can not scale application because there's already a scaling operation in progress")
```

Comment:
Start with lower case.
This isn't a NotAssigned error

Reply:
There seems no suitable error type for this. I guess we can use `errors.New()`?